src/lib/tls/tls12/tls_channel_impl_12.cpp

```diff
@@ -526,19 +526,24 @@ void Channel_Impl_12::write_record(Connection_Cipher_State* cipher_state, uint16
 
    if(cipher_state == nullptr)
       {
-      TLS::write_unencrypted_record(m_writebuf, record_type, record_version, next_seq,
+      TLS::write_unencrypted_record(m_writebuf,
+                                    static_cast<uint8_t>(record_type),
+                                    record_version, next_seq,
                                     input, length);
       }
    else
       {
-      TLS::write_record(m_writebuf, record_type, record_version, next_seq,
+      TLS::write_record(m_writebuf,
+                        static_cast<uint8_t>(record_type),
+                        record_version, next_seq,
```

Comment:
Why not adapt `write[_unprotected]_record` with the `Record_Type` parameter as well?

Reply:
I was lazy ;) done now